### PR TITLE
[Fix] Executing install.sh failed when ${password} and others contain @

### PR DESCRIPTION
**Pull request by tracehh**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Comment by tracehh** [comment]
1. sed -i ${txt} "s@^spring.datasource.password=.*@spring.datasource.password=${password}@g" conf/datasource.properties
2. ${password}  is qwer@1234
Then one more @

while  sed -i ${txt} "s/^spring.datasource.password=.*/spring.datasource.password=${password}/g"  and  ${password}  is  qwer/1234, the same error

May have a better way

**Comment by tracehh** [comment]
Each file is configured individually during deployment may be avoided.

File: install.sh [hunk 1]
```diff
@@ -41,8 +41,8 @@ sed -i ${txt} "s@^export JAVA_HOME=.*@export JAVA_HOME=${javaHome}@g" conf/env/d
 # Change configuration in conf/datasource.properties
 sed -i ${txt} "s@^spring.datasource.driver-class-name=.*@spring.datasource.driver-class-name=${datasourceDriverClassname}@g" conf/datasource.properties
 sed -i ${txt} "s@^spring.datasource.url=.*@spring.datasource.url=jdbc:${dbtype}://${dbhost}/${dbname}?characterEncoding=UTF-8\&allowMultiQueries=true@g" conf/datasource.properties
-sed -i ${txt} "s@^spring.datasource.username=.*@spring.datasource.username=${username}@g" conf/datasource.properties
-sed -i ${txt} "s@^spring.datasource.password=.*@spring.datasource.password=${password}@g" conf/datasource.properties
+sed -i ${txt} "s/^spring.datasource.username=.*/spring.datasource.username=${username}/g" conf/datasource.properties
+sed -i ${txt} "s/^spring.datasource.password=.*/spring.datasource.password=${password}/g" conf/datasource.properties
```
Comment (zhongjiajie) [inline install.sh]: Good catch! But when we password contains `/`, does this command work?

Comment (zhongjiajie) [reply install.sh]: I think maybe we should find more general way to escape specific characters

Comment (tracehh) [reply install.sh]: Yeah，need to consider both of  's','g','@'

Comment (zhongjiajie) [reply install.sh]: Yes, we have to consider other character that must be escaped. maybe we should add a escape function to do that, WDYT? @tracehh

Comment (tracehh) [reply install.sh]: Yes，I am working on this

**Comment by lenboo** [comment]
@tracehh Is there an issue associated with this PR

**Comment by zhongjiajie** [comment]
> @tracehh Is there an issue associated with this PR

@tracehh Yeah, maybe you should create a issue and target PR to it.

**Comment by tracehh** [comment]
ok, I create an issue : https://github.com/apache/dolphinscheduler/issues/6781

I open another pr for this issue. @zhongjiajie

**Comment by zhongjiajie** [comment]
> ok, I create an issue : #6781
> 
> I open another pr for this issue. @zhongjiajie

@tracehh Hi, you could just edit PR describe instead of open another PR, cause another PR would lost context about preview review. 
I could reopen PR and edit describe like screenshot belove

image

**Pull request closed without merging**